File: pyround1/1518.换酒问题.py
#
# @lc app=leetcode.cn id=1518 lang=python3
#
# [1518] 换酒问题
#

# @lc code=start
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        total = numBottles
        while numBottles >= numExchange:
            time = numBottles // numExchange
            numBottles = numBottles % numExchange + time
            total += time
        return total

# @lc code=end

File: 1518.换酒问题.cpp
/*
 * @lc app=leetcode.cn id=1518 lang=cpp
 *
 * [1518] 换酒问题
 */

// @lc code=start
class Solution {
public:
    int numWaterBottles(int numBottles, int numExchange) {
        int count = numBottles;
        int bottle = numBottles;
        while(bottle >= numExchange){
            bottle -= numExchange;
            count++;
            bottle++;
        }
        return count;

    }
};
// @lc code=end

